common/vm_wait_primary_nic_mac.yml

```diff
@@ -16,9 +16,25 @@
   until:
     - gather_network_facts.network_data is defined
     - gather_network_facts.network_data['0'] is defined
-    - gather_network_facts.network_data['0'].mac_addr is defined
-    - gather_network_facts.network_data['0'].mac_addr != ""
-  retries: ((300 | int) / 5 ) | int
+    - gather_network_facts.network_data['0'].mac_addr | bool
+  retries: 120
   delay: 5
   ignore_errors: true
   changed_when: false
+
+- name: "Check VM's primary network adapter have mac address"
+  ansible.builtin.assert:
+    that:
+      - gather_network_facts.network_data is defined
+      - gather_network_facts.network_data['0'] is defined
```

Comment:
Still need to check gather_network_facts.network_data['0'].mac_addr is defined.

Reply:
Thanks. Done.